magento/pwa-studio
# MDVA-39577: URL Rewrite is not working properly on Venia PWA

## Description
Steps:

1. Open Magento admin panel
2. Create a simple product with URL simple123.html
3. Marketing > SEO & Search > URL Rewrites
4. Create a new redirect:
- Request Path = natural-da-terra/products/123-raspberry-test
- Target Path = simple123.html
- Redirect Type = 302
5. Open Magento frontend http://pwa_domain/natural-da-terra/products/123-raspberry-test you will be redirected to  http://magento_domain/simple123.html
6. Open PWA frontend http://pwa_domain/natural-da-terra/products/123-raspberry-test

Actual result:
redirect to  http://pwa_domain/natural-da-terra/products/simple123.html

Expected result:
redirect to  http://pwa_domain/simple123.html
All path **natural-da-terra/products/123-raspberry-test** should be redirected to **simple123.html**

Case of the issue Magento does not use "/" in front of redirected URLs. In case if you will add "/" to the Target Path on step 4 Magento frontend redirect will work incorrectly. Result of redirect will be http://magento_domain//simple123.html

## Related Issue







Closes https://jira.corp.magento.com/browse/MDVA-39577.

## Acceptance

302 redirect should work correctly

### Verification Stakeholders



### Specification




## Verification Steps

1. Open Magento admin panel
2. Create a simple product with URL simple123.html
3. Marketing > SEO & Search > URL Rewrites
4. Create a new redirect:
- Request Path = natural-da-terra/products/123-raspberry-test
- Target Path = simple123.html
- Redirect Type = 302
5. Open Magento frontend http://pwa_domain/natural-da-terra/products/123-raspberry-test you will be redirected to  http://magento_domain/simple123.html
6. Open PWA frontend http://pwa_domain/natural-da-terra/products/123-raspberry-test

Actual result:
redirect to  http://pwa_domain/natural-da-terra/products/simple123.html

Expected result:
redirect to  http://pwa_domain/simple123.html
All path **natural-da-terra/products/123-raspberry-test** should be redirected to **simple123.html**


### Resolved issues:
1. [x] resolves magento/pwa-studio#3311: MDVA-39577: URL Rewrite is not working properly on Venia PWA
